### PR TITLE
[WIP][v6r19] Fix StorageElement inheritance and add a new Resources section 

### PR DESCRIPTION
**Pull request by aldbr**
This pull request: 

+ fixes different weird behaviours in Storage Element, as the inheritance of the protocol sections
+ creates a new Resources section named *StorageElementBases* containing only Base SEs
+ updates the documentation

### CODE REVIEW

**Comment by chaen** [comment]
Please put it WIP, we'll discuss it first

File: Resources/Storage/test/Test_StorageElement.py [hunk 1]
```diff
@@ -81,6 +81,79 @@ def mock_StorageFactory_generateStorageObject( storageName, pluginName, paramete
 
   return S_OK( storageObj )
 
+def mock_StorageFactory_getConfigStorageName( storageName, referenceType, rootConfigPath ):
```
Comment (fstagni) [inline Resources/Storage/test/Test_StorageElement.py]: There's no difference between this one and the one in Test_FilePlugin.py, right?

File: Resources/Storage/test/Test_StorageElement.py [hunk 2]
```diff
@@ -81,6 +81,79 @@ def mock_StorageFactory_generateStorageObject( storageName, pluginName, paramete
...
+  return S_OK( storageName )
+
+def mock_StorageFactory_getConfigStorageOptions( storageName, rootConfigPath, derivedStorageName = None ):
+  """ Get the options associated to the StorageElement as defined in the CS
+  """
+
+  options = {'BackendType': 'local',
+                            'ReadAccess': 'Active',
+                            'WriteAccess': 'Active',
```
Comment (fstagni) [inline Resources/Storage/test/Test_StorageElement.py]: Wrong indentations. Also, do you need this one when there's already a `mock_StorageFactory_getConfigStorageOptions` in Test_FilePlugin.py?

File: Resources/Storage/test/Test_StorageElement.py [hunk 3]
```diff
@@ -81,6 +81,79 @@ def mock_StorageFactory_generateStorageObject( storageName, pluginName, paramete
...
+
+  if storageName in ( 'StorageE', ):
+    options['WriteProtocols'] = ['root', 'srm']
+
+
+
+
+  return S_OK( options )
+
+def mock_StorageFactory_getConfigStorageProtocols( storageName, rootConfigPath, derivedStorageName = None ):
```
Comment (fstagni) [inline Resources/Storage/test/Test_StorageElement.py]: This is also in Test_FilePlugin.py but I start to understand that you may need different ones. 

In general, anyway these are not mock objects, they are more simply "fake objects". 
And one more thing: if there's a mock object that can be used by several test modules, we tend to put them in a single module, e.g. see files mock_FC.py and mock_DM.py.

Comment (chaen) [inline Resources/Storage/test/Test_StorageElement.py]: Alexandre will redo the PR anyway.
But concerning that, I think we once need to really have discussion on the topic, but I highly doubt that one can have mock that works for several use cases. To be seen

**Pull request closed without merging**